spec/system/consumer/checkout/guest_spec.rb

```diff
@@ -292,4 +292,30 @@
       end
     end
   end
+
+  shared_examples "when a line item is out of stock" do |session, step|
+    context "as a #{session} user" do
+      let(:user) { create(:user) }
+      before do
+        variant.on_demand = false
+        variant.on_hand = 0
+        variant.save!
+
+        if session == "logged"
+          login_as(user)
+        end
```

Comment:
Instead of this conditional, you could just do the login in a context block outside of the shared examples.